Applications/DataExplorer/VtkVis/VtkImageDataToSurfacePointsFilter.cpp

```diff
@@ -0,0 +1,158 @@
+/**
+ * \file
+ * \copyright
+ * Copyright (c) 2012-2018, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+// ** INCLUDES **
+#include "VtkImageDataToSurfacePointsFilter.h"
+
+#include <algorithm>
+#include <vector>
+
+#include <vtkIdList.h>
+#include <vtkImageData.h>
+#include <vtkInformation.h>
+#include <vtkInformationVector.h>
+#include <vtkLine.h>
+#include <vtkObjectFactory.h>
+#include <vtkPointData.h>
+#include <vtkPolyData.h>
+#include <vtkSmartPointer.h>
+
+vtkStandardNewMacro(VtkImageDataToSurfacePointsFilter);
+
+VtkImageDataToSurfacePointsFilter::VtkImageDataToSurfacePointsFilter()
+    : PointsPerPixel(20)
+{
+}
+
+VtkImageDataToSurfacePointsFilter::~VtkImageDataToSurfacePointsFilter() = default;
+
+void VtkImageDataToSurfacePointsFilter::PrintSelf(ostream& os, vtkIndent indent)
+{
+    this->Superclass::PrintSelf(os, indent);
+}
+
+int VtkImageDataToSurfacePointsFilter::FillInputPortInformation(int, vtkInformation* info)
+{
+    info->Set(vtkAlgorithm::INPUT_REQUIRED_DATA_TYPE(), "vtkImageData");
+    return 1;
+}
+
+int VtkImageDataToSurfacePointsFilter::RequestData(
```

Comment:
Parts of the code look very similar to the implementation of `int VtkImageDataToPointCloudFilter::RequestData`; would it be possible to reduce the code duplication?

Reply:
Originally I wanted to put both algorithms into the same filter. But then I noticed that the actual algorithms are fairly different. The set of parameters is completely different. One filter needs density information, the other doesn't. One interpolates point positions using a function, the other using a raster. I know that it looks similar but it isn't really.